gene2phenotype_project/gene2phenotype_app/serializers.py

```diff
@@ -985,111 +988,114 @@ class Meta:
-        disease = validated_data.get('disease')
-        date_created = datetime.now()
-        date_reviewed = date_created
-        session_name = validated_data.get('session_name')
+        return data
 
-        # In the HTML form the session_name is a list
-        # Change it to a string
-        if isinstance(session_name, list):
-            if session_name[0] == "":
-                session_name = ""
-            else:
-                session_name = session_name[0]
+    def convert_to_dict(self, data):
+        """
+            Convert data to a regular dictionary if it is an OrderedDict.
 
-        # Build the JSON
-        # We should not use the 'validated_data' because the data insert by the HTML form could have a different format
-        data_json_input = CurationDataSerializer.format_json(validated_data)
+            Parameters:
+                data: Data to convert (can be OrderedDict or dict).
 
-        # Assuming the user is already validated in the view AddCurationData
-        user_email = self.context.get('user')
-        user_obj = User.objects.get(email=user_email)
+            Returns:
+                Converted data as a dict.
+            Reason:
+                If the data is an OrderedDict, which is how Python is reading the JSON, it is turned to a regular dictionary which is what is returned, otherwise it just returns the data gotten
+        """
+        if isinstance(data, OrderedDict):
+            return dict(data)
+        
+        return data
+    
+    #using the Deepdiff module, compare JSON data 
+    # this still needs to be worked on when we have fixed the user permission issue 
+    def compare_curation_data(self, input_json_data, user_obj):
+       """"
+            Function to compare provided JSON data against JSON data stored in CurationData instances associated with a specific user.
+            Only compares the first layer of JSON objects.
+                Parameters:
+                    input_json_data: JSON data to compare against.
+                    user_obj: User object whose associated CurationData instances are to be checked.
+                 Returns:
+                    If a match is found, returns the corresponding CurationData instance.
+                    If no match is found, returns None.
+        """
+       user_sessions_queryset = CurationData.objects.filter(user=user_obj)
+       for curation_data in user_sessions_queryset:
+            data_json = curation_data.json_data
+            input_json_data["json_data"].pop('session_name', None) # removed to allow for proper comparison with both of them
+            data_json.pop('session_name', None) # removed to allow for proper comparison with both of them
+            result = DeepDiff(input_json_data["json_data"], data_json) # to compare curation data
+            
+            if not result:
+                return curation_data
+    
+    def check_entry(self, input_json_data):
```

Comment:
This method is not being called. Is there a reason for that?

Reply:
Yeah did not make any changes to this bit of the  existing code, i think it is for publish part but will also be done differently, just added comments to it so I can remember to make changes to it in the future